### PR TITLE
chore: Add X-Goog-Api-Client header to Auth and FCM requests

### PR DESCRIPTION
**Pull request by lahirumaramba**
Will be included in `v4.15.0`

### CODE REVIEW

File: auth/auth.go [hunk 1]
```diff
@@ -137,6 +138,7 @@ func NewClient(ctx context.Context, conf *internal.AuthConfig) (*Client, error)
 	hc.CreateErrFn = handleHTTPError
 	hc.Opts = []internal.HTTPOption{
 		internal.WithHeader("X-Client-Version", fmt.Sprintf("Go/Admin/%s", conf.Version)),
+		internal.WithHeader("x-goog-api-client", fmt.Sprintf("gl-go/%s fire-admin/%s", runtime.Version(), conf.Version)),
```
Comment (jonathanedey) [inline auth/auth.go:141]: The output of `runtime.Version()` has a prefix. Example: `go1.21` https://pkg.go.dev/runtime#Version

Would this affect parsing in our regex?

**Review by jonathanedey — APPROVED** [review]
LGTM with one comment on getting the go version.

File: auth/auth.go (hunk 1 above)
Comment (lahirumaramba) [reply auth/auth.go:141]: Good catch! I think that should be okay as it will be consistent everywhere. Let's confirm this offline with Vinay. Thanks!

Comment (lahirumaramba) [reply auth/auth.go]: Removed the prefix. Thanks!

**Pull request merged**